src/quicktype-typescript-input/index.ts

```diff
@@ -18,93 +16,14 @@ const compilerOptions: ts.CompilerOptions = {
     module: ts.ModuleKind.CommonJS,
     strictNullChecks: true,
-    typeRoots: []
+    typeRoots: [],
+    rootDir: "."
 };
 
-const libFileName = "lib.d.ts";
-let libSource: string | undefined = undefined;
-
-function getLibSource(): string {
-    if (libSource === undefined) {
-        libSource = inflateBase64(encodedDefaultTypeScriptLibrary);
-    }
-    return libSource;
-}
-
-class CompilerHost implements ts.CompilerHost {
-    constructor(_options: ts.CompilerOptions, private readonly _sources: { [fileName: string]: string }) {}
-
-    fileExists(fileName: string): boolean {
-        if (fileName === libFileName) return true;
-        return Object.prototype.hasOwnProperty.call(this._sources, fileName);
-    }
-
-    readFile(fileName: string): string | undefined {
-        if (fileName === libFileName) {
-            return getLibSource();
-        }
-        return this._sources[fileName];
-    }
-
-    getSourceFile(
-        fileName: string,
-        languageVersion: ts.ScriptTarget,
-        _onError?: (message: string) => void,
-        _shouldCreateNewSourceFile?: boolean
-    ): ts.SourceFile | undefined {
-        const sourceText = this.readFile(fileName);
-        return sourceText !== undefined ? ts.createSourceFile(fileName, sourceText, languageVersion) : undefined;
-    }
-
-    getDefaultLibFileName(_options: CompilerOptions): string {
-        return libFileName;
-    }
-
-    writeFile(fileName: string): void {
-        return panic(`writeFile should not be called by the TypeScript compiler.  Filename ${fileName}`);
-    }
-
-    getCurrentDirectory(): string {
-        return ".";
-    }
-
-    getDirectories(_path: string): string[] {
-        return [];
-    }
-
-    getCanonicalFileName(fileName: string): string {
-        if (this.useCaseSensitiveFileNames()) {
-            return fileName.toLowerCase();
-        }
-        return fileName;
-    }
-
-    useCaseSensitiveFileNames(): boolean {
-        return false;
-    }
-
-    getNewLine(): string {
-        return "\n";
-    }
-}
-
 // FIXME: We're stringifying and then parsing this schema again.  Just pass around
 // the schema directly.
-export function schemaForTypeScriptSources(sourceFileNames: string[]): JSONSchemaSourceData;
-export function schemaForTypeScriptSources(sources: { [fileName: string]: string }): JSONSchemaSourceData;
-export function schemaForTypeScriptSources(sources: string[] | { [fileName: string]: string }): JSONSchemaSourceData {
-    let fileNames: string[];
-    let host: ts.CompilerHost;
-
-    if (Array.isArray(sources)) {
-        fileNames = sources;
-        host = ts.createCompilerHost(compilerOptions);
-    } else {
-        fileNames = Object.getOwnPropertyNames(sources);
-        host = new CompilerHost(compilerOptions, sources);
-    }
-
-    const program = ts.createProgram(fileNames, compilerOptions, host);
+export function schemaForTypeScriptSources(sourceFileNames: string[]): JSONSchemaSourceData {
```

Comment:
Removing the `sources: { [fileName: string]: string }` argument overload breaks the quicktype web app.

Reply:
No longer used in web app